### PR TITLE
re-enabling auth for rx

### PR DESCRIPTION
**Pull request by saneshark**

### CODE REVIEW

**Comment by saneshark** [comment]
Went ahead and DRYed it up for now. I think concerns is easier than updating a bunch of controllers that inherit, but I can revisit later if necessary.

File: app/controllers/concerns/mhv_controller_concerns.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    before_action :authorize
+    before_action :authenticate_client
+  end
+
+  def authorize
+    current_user&.can_access_mhv? || raise_access_denied
+  end
+
+  def raise_access_denied
+    raise Common::Exceptions::Forbidden, detail: 'You do not have access to prescriptions'
```
Comment (aub) [inline app/controllers/concerns/mhv_controller_concerns.rb]: I think you want this message to either be custom per controller or more generic.

Comment (saneshark) [inline app/controllers/concerns/mhv_controller_concerns.rb]: oh, good catch

**Comment by aub** [comment]
Merge away once it clears checks.

**Pull request merged**